pages/docs/tutorials/generating-code.md

```diff
@@ -0,0 +1,153 @@
+
+1. Install [Node.js](https://nodejs.org/en/download/) (v15 or newer).
+
+2. Install Git on your Operating System:
+    - [Install Git on MacOs](https://git-scm.com/download/mac)
+    - [Install Git on Windows](https://git-scm.com/download/win)
+    - [Install Git on Linux](https://git-scm.com/download/linux)
+
+# Generating code
+
+In this section, we will go ahead with the steps to enable us generate our code from the document;
+
+### 1. Install the generator to use it as a command-line(CLI) tool
+<CodeBlock language="bash">
+{`npm install -g @asyncapi/generator`}
+</CodeBlock>
+
+The command above helps install the generator tool globally.
+
+### 2. Create a directory for your projects and enter it:
+<CodeBlock language="bash">
+{`mkdir streetlights && cd "$_"`}
+</CodeBlock>
+
+### 3. Create a file with the AsyncAPI machine-readable description you defined before using terminal. 
+The `cat` command below is a utility command in Linux. On Windows use `type` instead of `cat`:
+<CodeBlock language="yaml">
+{`cat <<EOT >> asyncapi.yaml
+asyncapi: '2.5.0'
+info:
+  title: Streetlights API
+  version: '1.0.0'
+  description: |
+    The Smartylighting Streetlights API allows you
+    to remotely manage the city lights.
+  license:
+    name: Apache 2.0
+    url: 'https://www.apache.org/licenses/LICENSE-2.0'
+servers:
+  mosquitto:
+    url: mqtt://test.mosquitto.org
+    protocol: mqtt
+channels:
+  light/measured:
+    publish:
+      summary: Inform about environmental lighting conditions for a particular streetlight.
+      operationId: onLightMeasured
+      message:
+        name: LightMeasured
+        payload:
+          type: object
+          properties:
+            id:
+              type: integer
+              minimum: 0
+              description: Id of the streetlight.
+            lumens:
+              type: integer
+              minimum: 0
+              description: Light intensity measured in lumens.
+            sentAt:
+              type: string
+              format: date-time
+              description: Date and time when the message was sent.
+EOT`}
+</CodeBlock>
+
+### 4. Trigger generation of the Node.js code:
+<CodeBlock language="bash">
+{`ag asyncapi.yaml @asyncapi/nodejs-template -o output -p server=mosquitto`}
+</CodeBlock>
+
+From the command above, the `-o` specifies where to output the result.
+
+### 5. And voilà! List all files in directory and check that Node.js application is generated:
+<CodeBlock language="bash">
+{`cd output && ls`}
+</CodeBlock>
+
+# Running your code
+
+### 1. Install dependencies of newly generated application:
+<CodeBlock language="bash">
+{`npm install`}
+</CodeBlock>
+
+### 2. Start the application:
+<CodeBlock language="bash">
+{`npm start`}
+</CodeBlock>
+
+### 3. In another terminal install the MQTT.js library:
+<CodeBlock language="bash">
+{`npm install mqtt -g`}
+</CodeBlock>
+
+### 4. Send a correct message to your application:
```

Comment:
about 4 and 5, these are about sending good and bad message, just focus on one, not calling it `correct`, because the 4th tutorial is about message validation